In [12]:
import os
import numpy as np
import nibabel as nib
import random
import shutil

## 1. Check centroids of pituitary glands

In [2]:
img_path = '../data/kopf_SHIP/Kopf_t1_mpr_tra_iso_p2/'
label_path = '../data/kopf_SHIP/Kopf_t1_mpr_tra_iso_p2_label/'

In [5]:
img = nib.load(f'{img_path}/2025500102352.nii.gz')
img = img.get_fdata()
print(img.shape)

(176, 256, 176)


### 1.1. Split data

In [10]:
test = []
files = os.listdir(label_path)
for i in range(10):
    ind = random.randint(0,len(files)-1)
    test.append(files[ind])

In [11]:
test

['2025500102741.nii',
 '2025500104030.nii',
 '2025500103561.nii',
 '2025500102785.nii',
 '2025500102387.nii',
 '2025500103850.nii',
 '2025500103649.nii',
 '2025500103807.nii',
 '2025500104069.nii',
 '2025500103123.nii']

In [15]:
output = '../data/kopf_SHIP/test'
for file in test:
    shutil.move(f'{img_path}/{file}.gz',f'{output}/images/{file}.gz')
    shutil.move(f'{label_path}/{file}',f'{output}/labels/{file}')

### 1.2. Check centroids and sizes of pituitary glands

In [13]:
centroids = []
sizes = []
files = []
for file in os.listdir(label_path):
    files.append(file)
    labels = nib.load(os.path.join(label_path,file)).get_fdata()
    x, y, z = np.where(labels > 0)
    center = [np.mean(x),np.mean(y),np.mean(z)]
    size = [np.max(x)-np.min(x),np.max(y)-np.min(y),np.max(z) - np.min(z)]
    centroids.append(center)
    sizes.append(size)
centroids = np.array(centroids)
sizes = np.array(sizes)

In [17]:
centroids

array([[ 89.53466872, 136.54083205,  73.60862866],
       [ 85.73561644, 133.42876712,  64.8369863 ],
       [ 90.75206612, 135.3742621 ,  77.61275089],
       [ 88.31490015, 136.84178187,  71.51612903],
       [ 88.28779599, 145.15482696,  67.75591985],
       [ 86.8348135 , 132.26642984,  62.1634103 ],
       [ 89.81039755, 137.10244648,  54.97553517],
       [ 88.67058824, 137.58151261,  65.25378151],
       [ 88.12      , 140.0075    ,  64.5075    ],
       [ 89.15625   , 131.69293478,  59.35869565],
       [ 89.08986616, 133.40344168,  58.4665392 ],
       [ 86.88853503, 128.05626327,  67.34819533],
       [ 91.75335121, 129.81233244,  72.02144772],
       [ 86.81991525, 133.23305085,  66.50847458],
       [ 86.55898123, 134.81903485,  58.15683646],
       [ 89.75555556, 136.35555556,  74.19444444],
       [ 88.56428571, 138.38452381,  69.45595238],
       [ 86.9954955 , 131.92642643,  70.01801802],
       [ 88.42668428, 137.75825627,  63.25231176],
       [ 87.203125  , 136.44687

In [16]:
sizes

array([[13, 10,  6],
       [16, 11,  6],
       [12, 13,  8],
       [13, 11,  8],
       [11, 13,  6],
       [15, 13,  6],
       [12, 12,  7],
       [14, 11,  6],
       [11,  8,  5],
       [17, 11,  8],
       [12,  9,  5],
       [18, 11,  8],
       [10,  8,  6],
       [15,  8,  5],
       [11, 12,  7],
       [10,  7,  2],
       [13, 11,  7],
       [16, 11,  6],
       [16, 13,  7],
       [13, 10,  6],
       [14, 11,  6],
       [12, 11,  8],
       [13,  7,  5],
       [11,  8,  4],
       [15, 10,  6],
       [15,  8,  7],
       [13,  9,  5],
       [14, 10,  7],
       [12, 11,  7],
       [13, 11, 10],
       [14, 10,  5],
       [15, 10,  5],
       [ 8, 10,  5],
       [12, 11,  8],
       [14, 11,  9],
       [10,  7,  6],
       [13,  9,  6],
       [15,  9,  6],
       [14, 10,  6],
       [12, 10,  7],
       [12, 11,  6],
       [11, 10,  7],
       [10, 11,  7],
       [11, 10,  8],
       [12, 10,  7],
       [ 9, 13,  8],
       [12,  7,  6],
       [ 8, 1

In [22]:
sizes.shape

(110, 3)

In [17]:
np.where(sizes[:,2] == 2)

(array([15]),)

In [18]:
files[15]

'2025500103466.nii'

In [20]:
coordinates = np.mean(centroids,axis = 0)
print(coordinates)

[ 88.27976135 134.8566023   68.66767387]


In [21]:
avg_size = np.mean(sizes,axis = 0)
print(avg_size)

[12.99090909 10.52727273  6.56363636]


### Central coordinates
Chosen coordinates are x = 88, y = 135, z = 69. The size of pituitary gland on average is x_length = 12.99mm, y_length = 10.53mm, z = 6.56mm. The cube size of 32 x 32 x 32 should cover the whole region of interest